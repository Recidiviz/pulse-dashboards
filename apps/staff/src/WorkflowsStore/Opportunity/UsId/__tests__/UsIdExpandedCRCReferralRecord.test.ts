// Recidiviz - a data platform for criminal justice reform
// Copyright (C) 2024 Recidiviz, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
// =============================================================================

import {
  UsIdExpandedCRCReferralRecordRaw,
  usIdExpandedCRCSchema,
} from "../UsIdExpandedCRCOpportunity";

test("transforms record with FTCD and PED set", () => {
  const rawRecord: UsIdExpandedCRCReferralRecordRaw = {
    stateCode: "US_MI",
    externalId: "xcrc-eligible-01",
    eligibleCriteria: {
      custodyLevelIsMinimum: {
        custodyLevel: "MINIMUM",
      },
      notServingForSexualOffense: null,
      usIdNoDetainersForXcrcAndCrc: null,
      usIdNoAbsconsionEscapeAndEludingPoliceOffensesWithin10Years: null,
      usIdIncarcerationWithin6MonthsOfFtcdOrPedOrTpd: {
        fullTermCompletionDate: "2022-08-12",
        paroleEligibilityDate: "2022-03-19",
        tentativeParoleDate: null,
      },
      usIdInCrcFacilityOrPwccUnit1: {
        crcStartDate: "2022-02-06",
        facilityName: "Facility 73",
      },
      usIdInCrcFacilityOrPwccUnit1For60Days: {
        sixtyDaysInCrcFacilityDate: "2022-04-07",
      },
    },
    ineligibleCriteria: {},
    isEligible: true,
    isAlmostEligible: false,
  };

  expect(usIdExpandedCRCSchema.parse(rawRecord)).toMatchSnapshot();
});

test("transforms record with TPD set", () => {
  const rawRecord: UsIdExpandedCRCReferralRecordRaw = {
    stateCode: "US_MI",
    externalId: "xcrc-eligible-02",
    eligibleCriteria: {
      custodyLevelIsMinimum: {
        custodyLevel: "MINIMUM",
      },
      notServingForSexualOffense: null,
      usIdNoDetainersForXcrcAndCrc: null,
      usIdNoAbsconsionEscapeAndEludingPoliceOffensesWithin10Years: null,
      usIdIncarcerationWithin6MonthsOfFtcdOrPedOrTpd: {
        fullTermCompletionDate: null,
        paroleEligibilityDate: null,
        tentativeParoleDate: "2022-07-11",
      },
      usIdInCrcFacilityOrPwccUnit1: {
        crcStartDate: "2022-02-06",
        facilityName: "Vault 6",
      },
      usIdInCrcFacilityOrPwccUnit1For60Days: {
        sixtyDaysInCrcFacilityDate: "2022-04-07",
      },
    },
    ineligibleCriteria: {},
    isEligible: true,
    isAlmostEligible: false,
  };

  expect(usIdExpandedCRCSchema.parse(rawRecord)).toMatchSnapshot();
});

test("transforms record with caseNotes set", () => {
  const rawRecord: UsIdExpandedCRCReferralRecordRaw = {
    stateCode: "US_MI",
    externalId: "xcrc-eligible-02",
    eligibleCriteria: {
      custodyLevelIsMinimum: {
        custodyLevel: "MINIMUM",
      },
      notServingForSexualOffense: null,
      usIdNoDetainersForXcrcAndCrc: null,
      usIdNoAbsconsionEscapeAndEludingPoliceOffensesWithin10Years: null,
      usIdIncarcerationWithin6MonthsOfFtcdOrPedOrTpd: {
        fullTermCompletionDate: null,
        paroleEligibilityDate: null,
        tentativeParoleDate: "2022-07-11",
      },
      usIdInCrcFacilityOrPwccUnit1: {
        crcStartDate: "2022-02-06",
        facilityName: "Vault 6",
      },
      usIdInCrcFacilityOrPwccUnit1For60Days: {
        sixtyDaysInCrcFacilityDate: "2022-04-07",
      },
    },
    ineligibleCriteria: {},
    caseNotes: {
      SomeSection: [
        {
          noteBody: "Some pig!",
        },
      ],
    },
    isEligible: true,
    isAlmostEligible: false,
  };

  expect(usIdExpandedCRCSchema.parse(rawRecord)).toMatchSnapshot();
});
