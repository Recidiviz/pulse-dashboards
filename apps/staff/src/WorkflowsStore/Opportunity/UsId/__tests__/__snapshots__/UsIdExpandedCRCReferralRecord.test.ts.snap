// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transforms record with FTCD and PED set 1`] = `
Object {
  "caseNotes": Object {},
  "eligibleCriteria": Object {
    "custodyLevelIsMinimum": Object {
      "custodyLevel": "MINIMUM",
    },
    "notServingForSexualOffense": null,
    "usIdInCrcFacilityOrPwccUnit1": Object {
      "crcStartDate": 2022-02-06T00:00:00.000Z,
      "facilityName": "Facility 73",
    },
    "usIdInCrcFacilityOrPwccUnit1For60Days": Object {
      "sixtyDaysInCrcFacilityDate": 2022-04-07T00:00:00.000Z,
    },
    "usIdIncarcerationWithin6MonthsOfFtcdOrPedOrTpd": Object {
      "fullTermCompletionDate": 2022-08-12T00:00:00.000Z,
      "paroleEligibilityDate": 2022-03-19T00:00:00.000Z,
      "tentativeParoleDate": null,
    },
    "usIdNoAbsconsionEscapeAndEludingPoliceOffensesWithin10Years": null,
    "usIdNoDetainersForXcrcAndCrc": null,
  },
  "externalId": "xcrc-eligible-01",
  "ineligibleCriteria": Object {},
  "stateCode": "US_MI",
}
`;

exports[`transforms record with TPD set 1`] = `
Object {
  "caseNotes": Object {},
  "eligibleCriteria": Object {
    "custodyLevelIsMinimum": Object {
      "custodyLevel": "MINIMUM",
    },
    "notServingForSexualOffense": null,
    "usIdInCrcFacilityOrPwccUnit1": Object {
      "crcStartDate": 2022-02-06T00:00:00.000Z,
      "facilityName": "Vault 6",
    },
    "usIdInCrcFacilityOrPwccUnit1For60Days": Object {
      "sixtyDaysInCrcFacilityDate": 2022-04-07T00:00:00.000Z,
    },
    "usIdIncarcerationWithin6MonthsOfFtcdOrPedOrTpd": Object {
      "fullTermCompletionDate": null,
      "paroleEligibilityDate": null,
      "tentativeParoleDate": 2022-07-11T00:00:00.000Z,
    },
    "usIdNoAbsconsionEscapeAndEludingPoliceOffensesWithin10Years": null,
    "usIdNoDetainersForXcrcAndCrc": null,
  },
  "externalId": "xcrc-eligible-02",
  "ineligibleCriteria": Object {},
  "stateCode": "US_MI",
}
`;

exports[`transforms record with caseNotes set 1`] = `
Object {
  "caseNotes": Object {
    "SomeSection": Array [
      Object {
        "noteBody": "Some pig!",
      },
    ],
  },
  "eligibleCriteria": Object {
    "custodyLevelIsMinimum": Object {
      "custodyLevel": "MINIMUM",
    },
    "notServingForSexualOffense": null,
    "usIdInCrcFacilityOrPwccUnit1": Object {
      "crcStartDate": 2022-02-06T00:00:00.000Z,
      "facilityName": "Vault 6",
    },
    "usIdInCrcFacilityOrPwccUnit1For60Days": Object {
      "sixtyDaysInCrcFacilityDate": 2022-04-07T00:00:00.000Z,
    },
    "usIdIncarcerationWithin6MonthsOfFtcdOrPedOrTpd": Object {
      "fullTermCompletionDate": null,
      "paroleEligibilityDate": null,
      "tentativeParoleDate": 2022-07-11T00:00:00.000Z,
    },
    "usIdNoAbsconsionEscapeAndEludingPoliceOffensesWithin10Years": null,
    "usIdNoDetainersForXcrcAndCrc": null,
  },
  "externalId": "xcrc-eligible-02",
  "ineligibleCriteria": Object {},
  "stateCode": "US_MI",
}
`;
